### PR TITLE
Feature/weheat integration

### PR DESCRIPTION
**Pull request by kjell-van-straaten**
## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/123057
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/5725
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Introduced Weheat integration for Home Assistant, allowing seamless management and display of Weheat devices.
	- Enhanced functionalities across Energy, Climate, and Sensor categories for improved user interaction.
	- Added comprehensive documentation for configuration and secure account linking, simplifying the integration process.

- **Improvements**
	- Implemented cloud polling features for a better overall user experience.

### CODE REVIEW

File: source/_integrations/weheat.markdown [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+title: Weheat
+description: Instructions on setting up Weheat within Home Assistant.
+ha_category:
+  - Energy
+  - Climate
+  - Sensor
+ha_iot_class: Cloud Polling
+ha_release: '2024.7.3'
```
Comment (joostlek) [inline source/_integrations/weheat.markdown]:
```suggestion
ha_release: "2024.10"
```

File: source/_integrations/weheat.markdown [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ha_dhcp: true
+ha_config_flow: true
+ha_quality_scale: silver
```
Comment (joostlek) [inline source/_integrations/weheat.markdown]:
```suggestion
ha_config_flow: true
```

File: source/_integrations/weheat.markdown [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ha_codeowners:
+  - '@jesperraemaekers'
+  - '@kjell-van-straaten'
+ha_domain: weheat
+ha_platforms:
+  - energy
+  - climate
+  - sensor
```
Comment (joostlek) [inline source/_integrations/weheat.markdown]:
```suggestion
ha_platforms:
  - sensor
```

File: source/_integrations/weheat.markdown [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  - '@jesperraemaekers'
+  - '@kjell-van-straaten'
+ha_domain: weheat
+ha_platforms:
+  - sensor
+ha_integration_type: integration
+---
+
+The **Weheat** {% term integration %} allows you to display your [Weheat](https://www.weheat.nl/) devices through Home Assistant.
+
```
Comment (c0ffeeca7) [inline source/_integrations/weheat.markdown:21]:
```suggestion

## Prerequisites

- You need a Weheat account before you can use this integration.
- When adding the integration, you will be prompted to enter a **Name**, **Client ID**,  and **Client Secret**. The name is arbitrary, the ID and secret are provided in the [knowledge base](https://support.weheat.nl/s/article/Is-er-een-offici%C3%ABle-Home-Assistant-integratie).
- You need you Weheat **username** and **password**.
 
```

File: source/_integrations/weheat.markdown [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+1. Go to your instance of Home Assistant.
+2. Navigate to **Settings**.
+3. Go to **Devices & Services**.
+4. Click the **+ Add Integration** button.
+5. Search for **Weheat**.
+6. You will be prompted to fill in a **Name**, **Client ID**  and **Client Secret**. The name is arbitrary, the ID and secret are provided in the [knowledge base](https://support.weheat.nl/s/article/Is-er-een-offici%C3%ABle-Home-Assistant-integratie).
+7. You are now redirected to the Weheat login provider. Log in using your **username** and **password**.
+8. After login, press **link account** to link your account.
+9. Your device will automatically be added.
+10. You are all set and ready to go!
```
Comment (c0ffeeca7) [inline source/_integrations/weheat.markdown:35]:
```suggestion
{% include integrations/config_flow.md %}

```

Comment (c0ffeeca7) [inline source/_integrations/weheat.markdown]: Please add a section called ## Sensors, listing names, units, and descriptions of the sensors.

**Comment by c0ffeeca7** [comment]
I'm [drafting this PR](https://developers.home-assistant.io/docs/review-process#prs-are-being-drafted-when-changes-are-needed), as changes are needed. Please undraft it, once the changes have been implemented and the PR is ready for review again.

**Comment by c0ffeeca7** [comment]
@kjell-van-straaten, could you please take a look at the review comment, so that we can move this PR along?

File: source/_integrations/weheat.markdown (hunk 5 above)
Comment (c0ffeeca7) [inline source/_integrations/weheat.markdown]:
```suggestion
## Prerequisites

- You need a Weheat account, **username**, and **password**
- When adding the integration in Home Assistant, you will be prompted to enter a **Name**, **Client ID**,  and **Client Secret**.
  - The name is arbitrary, the ID and secret are provided in the [knowledge base](https://support.weheat.nl/s/article/Is-er-een-offici%C3%ABle-Home-Assistant-integratie).
- During setup of the integration in Home Assistant, you will be redirected to the Weheat login provider. Log in using your **username** and **password**.
  - After login, select **link account** to link your account.

{% include integrations/config_flow.md %}

## Entities

### Sensors

The Weheat integration provides the following sensors:

- **Output power**:
- **Input power**:
- **COP**:

```

Comment (c0ffeeca7) [reply source/_integrations/weheat.markdown]: this include config_flow element is s standardized section on adding an integration to Home Assistant
image

The elements that need to be available before adding the integration are listed in a Prerequisites section

Comment (c0ffeeca7) [reply source/_integrations/weheat.markdown]: I tried to add a section describing the sensors. please add a description and the units.

Comment (jesperraemaekers) [reply source/_integrations/weheat.markdown]: Thanks for the start, we'll fill it in now.

**Comment by jesperraemaekers** [comment]
I have completed the list, but on some integrations i see more a table style. @c0ffeeca7  what would be preferred?

**Comment by c0ffeeca7** [comment]
> I have completed the list, but on some integrations i see more a table style. @c0ffeeca7 what would be preferred?

Nice! Thank you for the docs. My personal preference is the non-table version because it currently works better on mobile.
image

versus

image

File: source/_integrations/weheat.markdown [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+## Entities
+
+### Sensors
+
+The Weheat integration provides the following sensors:
+
+- **Output power**: Thermal power added to the water in Watts
+- **Input power**: Electrical power in Watts
+- **COP**: The Coefficient of performance between the above two power measurements
+- **Water inlet temperature"**: The heat pump water inlet temperature in °C
```
Comment (c0ffeeca7) [inline source/_integrations/weheat.markdown]:
```suggestion
- **Water inlet temperature**: The heat pump water inlet temperature in °C
```

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @kjell-van-straaten :+1:

**Comment by c0ffeeca7** [comment]
In case you did not see this setting: Enabling [**Allow edits from maintainers**](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) lets maintainers edit your PRs and fix/push tiny tweaks onto your branch. 

If you did not enable this on purpose, that's fine, too.

**Comment by kjell-van-straaten** [comment]
Thanks for the tip, we will turn that on next time :)

**Pull request merged**